### PR TITLE
feat(tokens): add new tokens json files

### PR DESCRIPTION
**Pull request by BenOsodrac**
Issue number: internal

---------

## What is the new behavior?

- Added new structure of JSON files that are generated from Figma Variables
- Adjusted tokens.js script to correctly translate the new JSON format
- Adjusted tokens usage on components and CSS files
- Added html template to auto-generate html preview based on Design Tokens (wip)
- Updated snapshots, as some tokens values changed, specially colors.

## Does this introduce a breaking change?

- [x] Yes
- [ ] No

### CODE REVIEW

File: core/src/css/test/link/basic/link.e2e.ts-snapshots/link-standalone-ionic-md-ltr-light-Mobile-Chrome-linux.png
Comment (thetaPC) [inline core/src/css/test/link/basic/link.e2e.ts-snapshots/link-standalone-ionic-md-ltr-light-Mobile-Chrome-linux.png:1]: What caused these screenshots to be updated?

File: core/src/css/test/link/basic/link.e2e.ts-snapshots/link-underline-nested-ionic-md-ltr-light-Mobile-Chrome-linux.png
Comment (thetaPC) [inline core/src/css/test/link/basic/link.e2e.ts-snapshots/link-underline-nested-ionic-md-ltr-light-Mobile-Chrome-linux.png:1]: What caused these screenshots to be updated?

File: core/src/css/test/typography/basic/typography.e2e.ts-snapshots/ionic-semantic-global-styles-ionic-md-ltr-light-Mobile-Chrome-linux.png
Comment (thetaPC) [inline core/src/css/test/typography/basic/typography.e2e.ts-snapshots/ionic-semantic-global-styles-ionic-md-ltr-light-Mobile-Chrome-linux.png:1]: What caused these screenshots to be updated?

File: core/src/foundations/_design-tokens.json
Comment (thetaPC) [inline core/src/foundations/_design-tokens.json:1]: Why was this file deleted?

File: core/src/foundations/tokens/effect.styles.tokens.json [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+      {
+        "offsetX": "0px",
+        "offsetY": "3px",
+        "blur": "14px",
+        "spread": "0px",
+        "color": "#062b631f"
+      }
+    ]
+  }
+}
```
Comment (thetaPC) [inline core/src/foundations/tokens/effect.styles.tokens.json:154]:
```suggestion
}

```

File: core/src/foundations/tokens/manifest.json [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+      "text.styles.tokens.json"
+    ],
+    "effect": [
+      "effect.styles.tokens.json"
+    ],
+    "color": [
+      "color.styles.tokens.json"
+    ]
+  }
+}
```
Comment (thetaPC) [inline core/src/foundations/tokens/manifest.json:30]:
```suggestion
}

```

File: core/src/foundations/tokens/text.styles.tokens.json
Comment (thetaPC) [inline core/src/foundations/tokens/text.styles.tokens.json:1]: Where does this file get used?

**Review by thetaPC — CHANGES_REQUESTED** [review]

File: core/scripts/tokens.js [hunk 3]
```diff
@@ -1,73 +1,114 @@
...
 // eslint-disable-next-line @typescript-eslint/no-var-requires
 const StyleDictionary = require('style-dictionary');
 
 const { fileHeader } = StyleDictionary.formatHelpers;
-
-// Empty for as an example of how we can add some extra variables, not from the tokens JSON
-const customVariables = ``;
-
-// Prefix for all generated variables
+const customVariables = ''; // Empty for example of how we can add some extra variables, not from the tokens JSON
```
[the remark this answers is not in the document]
Comment (BenOsodrac) [reply core/scripts/tokens.js]: Ah good to know, thanks. This exact line I ended up removing on a recent commit, but will keep that im mind for other cases 👍

File: core/scripts/tokens.js [hunk 4]
```diff
@@ -1,73 +1,114 @@
...
+$${variablesPrefix}-${prop.name}: (
+  font-family: $ionic-font-family,
+  font-size: $ionic-font-size-${fontSizeMap[typography.fontSize]},
+  font-weight: $ionic-font-weight-${fontWeightMap[typography.fontWeight]},
+  letter-spacing: $ionic-letter-spacing-${letterSpacingMap[typography.letterSpacing] || 0},
+  line-height: $ionic-line-height-${lineHeightMap[typography.lineHeight]},
+  text-transform: ${typography.textTransform},
+  text-decoration: ${typography.textDecoration}
+);
+`;
```
[the remark this answers is not in the document]
Comment (BenOsodrac) [reply core/scripts/tokens.js]: This we can't change, because it needs to have those exact spacings, so that it compiles to the CSS with the expected lint rules.

File: core/src/foundations/tokens/color.styles.tokens.json [hunk 5]
```diff
@@ -0,0 +1,18 @@
+{
+  "Guidelines": {
+    "$type": "color",
+    "$value": "#9747ff"
+  },
+  "Disabled": {
+    "$type": "color",
+    "$value": "#ffffff99"
```
[the remark this answers is not in the document]
Comment (BenOsodrac) [reply core/src/foundations/tokens/color.styles.tokens.json:8]: Ah thanks for the heads-up, didn't know that! Will adjust the code 👍

File: core/src/foundations/tokens/color.styles.tokens.json [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+  },
+  "Hover": {
+    "$type": "color",
+    "$value": "#1212120a"
+  },
+  "Pressed": {
+    "$type": "color",
+    "$value": "#12121214"
+  }
+}
```
[the remark this answers is not in the document]
Comment (BenOsodrac) [reply core/src/foundations/tokens/color.styles.tokens.json:18]: I can change it, but these come directly from Figma, so we shouldn't change them directly!

File: core/src/foundations/tokens/manifest.json (hunk 2 above)
Comment (BenOsodrac) [reply core/src/foundations/tokens/manifest.json:30]: These files come directly from Figma, we shouldn't change them directly!

File: core/src/foundations/tokens/text.styles.tokens.json
Comment (BenOsodrac) [reply core/src/foundations/tokens/text.styles.tokens.json:1]: This is used on the `tokens.js` file to generate typography scss maps on `ionic.vars.scss` and utility-classes on `ionic.utility.scss`

File: core/src/foundations/_design-tokens.json
Comment (BenOsodrac) [reply core/src/foundations/_design-tokens.json:1]: Because it contained the old tokens values from the first version. Its replaced now with the new JSON files that come from Figma.

File: core/scripts/tokens.js [hunk 7]
```diff
@@ -1,73 +1,114 @@
...
+}
+
+function createTypeMap(dictionary, type) {
```
[the remark this answers is not in the document]
Comment (BenOsodrac) [reply core/scripts/tokens.js]: Added a comment for all functions 👍

File: core/src/components/card/card.ionic.scss [hunk 8]
```diff
@@ -4,15 +4,15 @@
...
+  --border-radius: #{globals.$ionic-border-radius-100};
 
-  @include globals.padding(globals.$ionic-space-base);
+  @include globals.padding(globals.$ionic-space-400);
   @include globals.border-radius(var(--border-radius));
 
   display: block;
 
-  border: #{globals.$ionic-border-size-small} solid #{globals.$ionic-color-neutral-50};
+  border: #{globals.$ionic-border-size-025} solid #{globals.$ionic-color-neutral-500};
```
[the remark this answers is not in the document]
Comment (BenOsodrac) [reply core/src/components/card/card.ionic.scss:15]: According to UX/UI team almost all color values changed (this had already warned about this, the first color draft was not final). These new tokens should be the final ones for a long time now.

For now, I just kept using the same equivalent tokens (10 is now 100, etc), that’s why we have differences on the components we did already, as all the colors changed a bit.

Regarding Figma component files showing different values on the tokens, I already contacted Beth Thomas about this, its probably because she didn’t yet propagated the changes on their side.

But the Design Tokens should always represent the source of truth!

File: core/src/components/chip/test/shape/chip.e2e.ts-snapshots/chip-default-ionic-md-ltr-light-Mobile-Chrome-linux.png
[the remark this answers is not in the document]
Comment (BenOsodrac) [reply core/src/components/chip/test/shape/chip.e2e.ts-snapshots/chip-default-ionic-md-ltr-light-Mobile-Chrome-linux.png:1]: See https://github.com/ionic-team/ionic-framework/pull/29525#discussion_r1607921941

File: core/src/components/tab-bar/test/basic/tab-bar.e2e.ts-snapshots/tab-bar-default-ionic-md-ltr-light-Mobile-Chrome-linux.png
[the remark this answers is not in the document]
Comment (BenOsodrac) [reply core/src/components/tab-bar/test/basic/tab-bar.e2e.ts-snapshots/tab-bar-default-ionic-md-ltr-light-Mobile-Chrome-linux.png:1]: See https://github.com/ionic-team/ionic-framework/pull/29525#discussion_r1607921941

File: core/src/css/ionic/core.ionic.scss [hunk 9]
```diff
@@ -4,7 +4,7 @@
 // --------------------------------------------------
 
 :root {
-  --ionic-global-background-color: #{globals.$ionic-color-neutral-10};
+  --ionic-global-background-color: #{globals.$ionic-color-neutral-100};
```
[the remark this answers is not in the document]
Comment (BenOsodrac) [reply core/src/css/ionic/core.ionic.scss:7]: See https://github.com/ionic-team/ionic-framework/pull/29525#discussion_r1607921941

File: core/src/css/test/link/basic/link.e2e.ts-snapshots/link-standalone-ionic-md-ltr-light-Mobile-Chrome-linux.png
Comment (BenOsodrac) [reply core/src/css/test/link/basic/link.e2e.ts-snapshots/link-standalone-ionic-md-ltr-light-Mobile-Chrome-linux.png:1]: See https://github.com/ionic-team/ionic-framework/pull/29525#discussion_r1607921941

File: core/src/css/test/link/basic/link.e2e.ts-snapshots/link-underline-nested-ionic-md-ltr-light-Mobile-Chrome-linux.png
Comment (BenOsodrac) [reply core/src/css/test/link/basic/link.e2e.ts-snapshots/link-underline-nested-ionic-md-ltr-light-Mobile-Chrome-linux.png:1]: See https://github.com/ionic-team/ionic-framework/pull/29525#discussion_r1607921941

File: core/src/css/test/typography/basic/typography.e2e.ts-snapshots/ionic-semantic-global-styles-ionic-md-ltr-light-Mobile-Chrome-linux.png
Comment (BenOsodrac) [reply core/src/css/test/typography/basic/typography.e2e.ts-snapshots/ionic-semantic-global-styles-ionic-md-ltr-light-Mobile-Chrome-linux.png:1]: See https://github.com/ionic-team/ionic-framework/pull/29525#discussion_r1607921941

File: core/scripts/tokens.js [hunk 10]
```diff
@@ -1,73 +1,114 @@
...
+    ? `$${variablesPrefix}-${propName}: var(--${variablesPrefix}-${propName}, "${prop.value}", sans-serif);`
+    : `--${variablesPrefix}-${propName}: "${prop.value}", sans-serif;`;
+}
+
+function generateTypographyValue(prop, dictionary) {
+  const typography = prop.value;
+  const fontSizeMap = createTypeMap(dictionary, 'font-size');
+  const fontWeightMap = createTypeMap(dictionary, 'font-weight');
+  const lineHeightMap = createTypeMap(dictionary, 'line-height');
+  const letterSpacingMap = createTypeMap(dictionary, 'letter-spacing');
```
[the remark this answers is not in the document]
Comment (BenOsodrac) [reply core/scripts/tokens.js]: Hum we are not creating the values, just getting them. 

Dictionary does have all the values, but each time we call this method, we need to iterate on the dictionary to get the respective ones for the current prop.

File: core/scripts/tokens.js [hunk 11]
```diff
@@ -131,6 +162,55 @@ StyleDictionary.registerFormat({
   },
 });
 
+function generateTypographyUtilityClass(prop, dictionary) {
+  const typography = prop.value;
+  const fontSizeMap = createTypeMap(dictionary, 'font-size');
+  const fontWeightMap = createTypeMap(dictionary, 'font-weight');
+  const lineHeightMap = createTypeMap(dictionary, 'line-height');
+  const letterSpacingMap = createTypeMap(dictionary, 'letter-spacing');
```
[the remark this answers is not in the document]
Comment (BenOsodrac) [reply core/scripts/tokens.js]: See https://github.com/ionic-team/ionic-framework/pull/29525#discussion_r1607955121

File: core/scripts/tokens.js [hunk 12]
```diff
@@ -131,6 +162,55 @@ StyleDictionary.registerFormat({
...
+    case 'weight':
+      fontAttribute = 'font-weight';
+      break;
+    case 'line-height':
+      fontAttribute = 'line-height';
+      break;
+    case 'letter-spacing':
+      fontAttribute = 'letter-spacing';
+      break;
+  }
```
[the remark this answers is not in the document]
Comment (BenOsodrac) [reply core/scripts/tokens.js]: I wouldn't say so, all tokens on JSON must have a type. if not, then it's an issue on the Figma Variables and the way they are generating the tokens!

File: core/scripts/tokens.js [hunk 13]
```diff
@@ -80,41 +121,31 @@ StyleDictionary.registerFormat({
...
+
       switch (tokenType) {
         case 'color':
-          utilityClass = `.${variablesPrefix}-${className} {\n  color: $ionic-${prop.name};\n}
-.${variablesPrefix}-background-${className} {\n  background-color: $ionic-${prop.name};\n}`;
+        case 'state':
+        case 'Guidelines':
+        case 'Disabled':
+        case 'Hover':
+        case 'Pressed':
```
[the remark this answers is not in the document]
Comment (BenOsodrac) [reply core/scripts/tokens.js]: Its matching the case from the JSON tokens. But I will double-check with Beth Thomas if she can create them all with the same case 👍

[the remark this answers is not in the document]
Comment (BenOsodrac) [reply core/scripts/tokens.js]: But, as of today, this is correct!

File: core/scripts/tokens.js [hunk 14]
```diff
@@ -80,41 +121,31 @@ StyleDictionary.registerFormat({
...
+      } else if (prop.attributes.category.match('font-family') || tokenType === 'scale') {
+        return;
+      }
+
       switch (tokenType) {
         case 'color':
-          utilityClass = `.${variablesPrefix}-${className} {\n  color: $ionic-${prop.name};\n}
-.${variablesPrefix}-background-${className} {\n  background-color: $ionic-${prop.name};\n}`;
+        case 'state':
+        case 'Guidelines':
```
[the remark this answers is not in the document]
Comment (BenOsodrac) [reply core/scripts/tokens.js]: Not sure yet, didn't stumble on any example that uses it. But we can ask UX team about it 👍

File: core/scripts/tokens.js [hunk 15]
```diff
@@ -1,73 +1,114 @@
...
-      const rgb = hexToRgb(prop.value);
-      return `$${variablesPrefix}-${prop.name}: var(--${variablesPrefix}-${prop.name}, ${prop.value});${
-        rgb
-          ? `\n$${variablesPrefix}-${prop.name}-rgb: var(--${variablesPrefix}-${prop.name}-rgb, ${rgb.r}, ${rgb.g}, ${rgb.b});`
-          : ``
-      }`;
+    const typographyProperties = dictionary.allProperties.filter((prop) => prop['$type'] === 'typography');
+    const otherProperties = dictionary.allProperties.filter((prop) => prop['$type'] !== 'typography');
+
+    const sortedProperties = [...otherProperties, ...typographyProperties];
```
[the remark this answers is not in the document]
Comment (BenOsodrac) [reply core/scripts/tokens.js]: To avoid compile errors due to undefined scss variables. Typography scss maps and utility-classes use other scss variables, so they must be defined first!

[the remark this answers is not in the document]
Comment (BenOsodrac) [reply core/scripts/tokens.js]: Added comment on that line 👍

File: core/scripts/tokens.js [hunk 16]
```diff
@@ -1,73 +1,114 @@
...
 function hexToRgb(hex) {
   const result = /^#?([a-f\d]{2})([a-f\d]{2})([a-f\d]{2})$/i.exec(hex);
-  return result
-    ? {
-        r: parseInt(result[1], 16),
-        g: parseInt(result[2], 16),
-        b: parseInt(result[3], 16),
-      }
-    : null;
+  return result ? { r: parseInt(result[1], 16), g: parseInt(result[2], 16), b: parseInt(result[3], 16) } : null;
```
[the remark this answers is not in the document]
Comment (BenOsodrac) [reply core/scripts/tokens.js]: Commited!

File: core/scripts/tokens.js (hunk 7 above)
[the remark this answers is not in the document]
Comment (BenOsodrac) [reply core/scripts/tokens.js]: Commited!

File: core/scripts/tokens.js [hunk 17]
```diff
@@ -1,73 +1,114 @@
...
+  );
+}
+
+function generateRgbValue(prop) {
+  const rgb = hexToRgb(prop.value);
+  return `$${variablesPrefix}-${prop.name}: var(--${variablesPrefix}-${prop.name}, ${prop.value});${
+    rgb
+      ? `\n$${variablesPrefix}-${prop.name}-rgb: var(--${variablesPrefix}-${prop.name}-rgb, ${rgb.r}, ${rgb.g}, ${rgb.b});`
+      : ``
+  }`;
```
[the remark this answers is not in the document]
Comment (BenOsodrac) [reply core/scripts/tokens.js]: Commited!

File: core/scripts/tokens.js (hunk 15 above)
[the remark this answers is not in the document]
Comment (BenOsodrac) [reply core/scripts/tokens.js]: Commited!

File: core/scripts/tokens.js [hunk 18]
```diff
@@ -80,41 +121,31 @@ StyleDictionary.registerFormat({
       const className = `${prop.name}`;
       let utilityClass = '';
 
+      if (tokenType.startsWith('Elevation')) {
+        return `.${variablesPrefix}-${className} {\n  box-shadow: $ionic-${prop.name};\n}`;
+      } else if (prop['$type'] === 'typography') {
+        return generateTypographyUtilityClass(prop, dictionary);
+      } else if (prop.attributes.category.match('font-family') || tokenType === 'scale') {
+        return;
```
[the remark this answers is not in the document]
Comment (BenOsodrac) [reply core/scripts/tokens.js]: Commited!

File: core/scripts/tokens.js [hunk 19]
```diff
@@ -131,6 +162,55 @@ StyleDictionary.registerFormat({
...
+      fontAttribute = 'letter-spacing';
+      break;
+  }
+  return `.${variablesPrefix}-${className} {\n  ${fontAttribute}: $ionic-${prop.name};\n}`;
+}
+
+function generateSpaceUtilityClasses(prop, className) {
+  return `.${variablesPrefix}-margin-${className} {\n  --margin-start: #{$ionic-${prop.name}};\n  --margin-end: #{$ionic-${prop.name}};\n  --margin-top: #{$ionic-${prop.name}};\n  --margin-bottom: #{$ionic-${prop.name}};\n\n  @include margin($ionic-${prop.name});\n};\n
+.${variablesPrefix}-padding-${className} {\n  --padding-start: #{$ionic-${prop.name}};\n  --padding-end: #{$ionic-${prop.name}};\n  --padding-top: #{$ionic-${prop.name}};\n  --padding-bottom: #{$ionic-${prop.name}};\n\n  @include padding($ionic-${prop.name});\n};\n`;
+}
```
[the remark this answers is not in the document]
Comment (BenOsodrac) [reply core/scripts/tokens.js]: Commited!

File: core/src/foundations/tokens/color.styles.tokens.json (hunk 5 above)
[the remark this answers is not in the document]
Comment (BenOsodrac) [reply core/src/foundations/tokens/color.styles.tokens.json:8]: Added a TODO for now, as I will need to tweak the logic that those color tokens are using. Will do it on the scope of this sprint's JIRA task 👍

**Comment by BenOsodrac** [comment]
@thetaPC @brandyscarney I did some recent commits to reorgazine the tokens.js code a bit better a separate some methods on a different file. Hopefully its a bit cleaner 👍

File: core/src/foundations/tokens/color.styles.tokens.json
Comment (thetaPC) [inline core/src/foundations/tokens/color.styles.tokens.json:1]: Can we add a README.md under this same folder, `core/src/foundations/tokens/`, saying that these JSON files are generated from Figma and should not be modified directly.

File: core/package.json [hunk 20]
```diff
@@ -77,7 +77,7 @@
     "build.css": "npm run css.sass && npm run css.minify",
     "build.debug": "npm run clean && stencil build --debug",
     "build.docs.json": "stencil build --docs-json dist/docs.json",
-    "build.tokens": "node ./scripts/tokens.js && npm run lint.sass.fix && npm run prettier.tokens",
+    "build.tokens": "node ./scripts/tokens/tokens.js && npm run lint.sass.fix && npm run prettier.tokens",
```
Comment (thetaPC) [inline core/package.json:80]: Nitpick: Let's call the file `index.js` since it's already in a `tokens` folder.

```suggestion
    "build.tokens": "node ./scripts/tokens/index.js && npm run lint.sass.fix && npm run prettier.tokens",
```

File: core/scripts/tokens/tokens-utilities.js [hunk 21]
```diff
@@ -0,0 +1,144 @@
+const variablesPrefix = 'ionic'; // varuable that holds the prefix used on all css and scss variables generated
```
Comment (thetaPC) [inline core/scripts/tokens/tokens-utilities.js:1]:
```suggestion
// Variable that holds the prefix used on all css and scss variables generated
const variablesPrefix = 'ionic';
```

File: core/scripts/tokens/tokens-utilities.js [hunk 22]
```diff
@@ -0,0 +1,144 @@
...
+  return result
+    ? {
+        r: parseInt(result[1], 16),
+        g: parseInt(result[2], 16),
+        b: parseInt(result[3], 16),
+      }
+    : null;
+}
+
+// Method to generated a valid box-shadow value from a shadow Design Token structure
```
Comment (thetaPC) [inline core/scripts/tokens/tokens-utilities.js:15]:
```suggestion
// Generates a valid box-shadow value from a shadow Design Token structure
```

File: core/scripts/tokens/tokens-utilities.js [hunk 23]
```diff
@@ -0,0 +1,144 @@
...
+// Method to generated a valid font-family value from a font-family Design Token structure
+function generateFontFamilyValue(prop, variableType = 'css') {
+  // Remove the last word from the token, as it contains the name of the font, which we don't want to be included on the generated variables
+  const propName = prop.name.split('-').slice(0, -1).join('-');
+  return variableType === 'scss'
+    ? `$${variablesPrefix}-${propName}: var(--${variablesPrefix}-${propName}, "${prop.value}", sans-serif);`
+    : `--${variablesPrefix}-${propName}: "${prop.value}", sans-serif;`;
+}
+
+// Method to generated a typography based scss map from a typography Design Token structure
```
Comment (thetaPC) [inline core/scripts/tokens/tokens-utilities.js:29]:
```suggestion
// Generates a typography based scss map from a typography Design Token structure
```

File: core/scripts/tokens/tokens-utilities.js [hunk 24]
```diff
@@ -0,0 +1,144 @@
...
+}
+
+// Method to be reused on the others methods of this file, to abstract the need to loop across all tokens from a given type
+function createTypeMap(dictionary, type) {
+  return Object.fromEntries(
+    Object.entries(dictionary.properties[type]).map(([key, token]) => [token.value, token.attributes.type])
+  );
+}
+
+// Method to generate a rgb color value, based on a color Design Token
```
Comment (thetaPC) [inline core/scripts/tokens/tokens-utilities.js:57]:
```suggestion
// Generates a rgb color value based on a color Design Token
```

File: core/scripts/tokens/tokens-utilities.js [hunk 25]
```diff
@@ -0,0 +1,144 @@
...
+
+  // If the token is color, also add a rgb variable using the same color
+  if (rgb) {
+    rgbDeclaration = `\n$${variablesPrefix}-${prop.name}-rgb: var(--${variablesPrefix}-${prop.name}-rgb, ${rgb.r}, ${rgb.g}, ${rgb.b});`;
+  }
+
+  return `$${variablesPrefix}-${prop.name}: var(--${variablesPrefix}-${prop.name}, ${prop.value});${rgbDeclaration}`;
+}
+
+// Method to generated a typography based css utility-class from a typography Design Token structure
```
Comment (thetaPC) [inline core/scripts/tokens/tokens-utilities.js:70]:
```suggestion
// Generates a typography based css utility-class from a typography Design Token structure
```

File: core/scripts/tokens/tokens-utilities.js [hunk 26]
```diff
@@ -0,0 +1,144 @@
...
+    font-weight: $ionic-font-weight-${fontWeightMap[typography.fontWeight]};
+    letter-spacing: $ionic-letter-spacing-${letterSpacingMap[typography.letterSpacing] || 0};
+    line-height: $ionic-line-height-${lineHeightMap[typography.lineHeight]};
+    text-transform: ${typography.textTransform};
+    text-decoration: ${typography.textDecoration};
+  };
+  `;
+}
+
+// Method to generated a color based css utility-class from a color Design Token structure
```
Comment (thetaPC) [inline core/scripts/tokens/tokens-utilities.js:91]:
```suggestion
// Generates a color based css utility-class from a color Design Token structure
```

**Review by thetaPC — CHANGES_REQUESTED** [review]
Thank you so much for all the comments!

File: core/scripts/tokens/preview-template.html
[the remark this answers is not in the document]
Comment (BenOsodrac) [reply core/scripts/tokens/preview-template.html:1]: I removed all code related to the html preview that was wip 👍

File: core/scripts/tokens/tokens.js
[the remark this answers is not in the document]
Comment (BenOsodrac) [reply core/scripts/tokens/tokens.js:1]: Hum weird I always use that, maybe I didn't noticed on some of the changes!

[the remark this answers is not in the document]
Comment (BenOsodrac) [reply core/scripts/tokens/tokens.js:1]: Done!

File: core/scripts/tokens/tokens.js [hunk 27]
```diff
@@ -0,0 +1,303 @@
...
+    const prefixedVariables = dictionary.allProperties
+      .filter((prop) => prop['$type'] !== 'typography')
+      .map((prop) => {
+        if (prop.attributes.category.startsWith('Elevation')) {
+          const cssShadow = prop.value.map(generateShadowValue).join(', ');
+          return `--${variablesPrefix}-${prop.name}: ${cssShadow};`;
+        } else if (prop.attributes.category.match('font-family')) {
+          return generateFontFamilyValue(prop);
+        } else {
+          // TODO(4870): prevent colors with 8 characters to be created without a rgb transformation
```
[the remark this answers is not in the document]
Comment (BenOsodrac) [reply core/scripts/tokens/tokens.js]: The ROU-4870 is a big ballpark where I will do a series of PR's on this sprint. On one of them I intend to implement that logic. 

We know is not ideal, but its a way to try to catch up some time lost, until we are on a more normal speed and state 😃

File: core/src/foundations/tokens/color.styles.tokens.json
Comment (BenOsodrac) [reply core/src/foundations/tokens/color.styles.tokens.json:1]: I think this is a bit overkill. The files itself already are generated with a header that has that exact info.

File: core/scripts/tokens/tokens.js
[the remark this answers is not in the document]
Comment (thetaPC) [reply core/scripts/tokens/tokens.js:1]: Oof, then it was one of those weird moments with git. That some times happens to me even though I used `git mv`.

File: core/scripts/tokens/index.js [hunk 28]
```diff
@@ -0,0 +1,214 @@
...
+    const prefixedVariables = dictionary.allProperties
+      .filter((prop) => prop['$type'] !== 'typography')
+      .map((prop) => {
+        if (prop.attributes.category.startsWith('Elevation')) {
+          const cssShadow = prop.value.map(generateShadowValue).join(', ');
+          return `--${variablesPrefix}-${prop.name}: ${cssShadow};`;
+        } else if (prop.attributes.category.match('font-family')) {
+          return generateFontFamilyValue(prop);
+        } else {
+          // TODO(4870): prevent colors with 8 characters to be created without a rgb transformation
```
Comment (thetaPC) [inline core/scripts/tokens/index.js:40]:
```suggestion
          // TODO(ROU-4870): prevent colors with 8 characters to be created without a rgb transformation
```

File: core/scripts/tokens.js (hunk 10 above)
[the remark this answers is not in the document]
Comment (thetaPC) [reply core/scripts/tokens.js]: @BenOsodrac why was this marked as resolved?

File: core/src/foundations/tokens/color.styles.tokens.json
Comment (thetaPC) [reply core/src/foundations/tokens/color.styles.tokens.json:1]: Yes, but the original files like `core/src/foundations/tokens/color.styles.tokens.json` can be modified before being generated. We should warn developers that any JSON file under `core/src/foundations/tokens/` should not be modified since they are files that come from the Figma. If changes are made then proceed with caution but the changes should be coming from the design team.

Thoughts?

Comment (BenOsodrac) [reply core/src/foundations/tokens/color.styles.tokens.json:1]: Fair enough, added README file 👍

File: core/scripts/tokens.js (hunk 10 above)
[the remark this answers is not in the document]
Comment (BenOsodrac) [reply core/scripts/tokens.js]: Sorry, my bad, was unintentional 🥲

[the remark this answers is not in the document]
Comment (thetaPC) [reply core/scripts/tokens.js]: Thank you!

**Review by thetaPC — APPROVED** [review]
LGTM, thank you!!
